### PR TITLE
Fix proto gen

### LINKED ISSUE #108358: hack/update-generated-protobuf.sh: no longer creates and/or updates generated.pb.go
### What happened?

I am trying to add a new API group and want to generate the `generated.pb.go` file for it with `hack/update-generated-protobuf.sh`. That fails in two ways:
- syncing out the result fails when the file does not exist yet
- if I create an empty Go file (just with `package` statement), that file doesn't get updated

### What did you expect to happen?

The `generated.pb.go` file should get updated. That works v1.22.0.

### How can we reproduce it (as minimally and precisely as possible)?

The first problem can be reproduced with current master (5bafc8306a7):

```console
$ rm staging/src/k8s.io/api/storage/v1alpha1/generated.pb.go
$ hack/update-generated-protobuf.sh
+++ [0225 22:15:27] Verifying Prerequisites....
+++ [0225 22:15:28] Building Docker image kube-build:build-1f83d3a10f-5-v1.24.0-go1.18rc1-bullseye.0
...
2022/02/25 22:21:19 Unable to rewrite generated /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/api/storage/v1alpha1/generated.pb.go: open /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/api/storage/v1alpha1/generated.pb.go: no such file or directory
```

### Anything else we need to know?

I bisected it with this script:
```
#!/bin/sh

rm -f staging/src/k8s.io/api/storage/v1alpha1/generated.pb.go
hack/update-generated-protobuf.sh
```

That points towards https://github.com/kubernetes/kubernetes/commit/1723bb1f79d69481f97d351218dc1b87c8ac1b60 as the first broken commit, from PR https://github.com/kubernetes/kubernetes/pull/106630.

/cc @howardjohn 

### Kubernetes version

Kubernetes master (5bafc8306a7).

### Cloud provider

n/a

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
$ uname -a
Linux pohly-desktop 5.10.0-0.bpo.9-amd64 #1 SMP Debian 5.10.70-1~bpo10+1 (2021-10-10) x86_64 GNU/Linux
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

https://github.com/kubernetes/kubernetes/issues/108358 made proto files contain full go import paths, but broke actual generation of the generated.pb.go files.

This PR fixes that by telling gogo to generate pb.go files next to the proto files when we're given full paths as input.

#### Which issue(s) this PR fixes:
Fixes #108358

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protoc-gen-gogo/main.go [hunk 1]
```diff
@@ -28,5 +30,18 @@ import (
 )
 
 func main() {
-	command.Write(command.Generate(command.Read()))
+	// read input
+	request := command.Read()
+
+	// if we're given paths as inputs, generate .pb.go files based on those paths
+	for _, file := range request.FileToGenerate {
+		if strings.Contains(file, "/") {
```
Comment (aojea) [inline staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protoc-gen-gogo/main.go:38]: Contains or prefix? `strings.HasPrefix(file, "/")`

Comment (liggitt) [reply staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protoc-gen-gogo/main.go:38]: contains. when given paths, it looks like `k8s.io/kubernetes/vendor/k8s.io/api/core/v1/generated.proto`

Comment (krmayankk) [reply staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protoc-gen-gogo/main.go:38]: should we put in the comment, an example of these expected paths to make it easier for other people looking at this code ?

File: staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go [hunk 2]
```diff
@@ -260,6 +260,7 @@ func Run(g *Generator) {
 	if len(g.Conditional) > 0 {
 		fmt.Fprintf(buf, "// +build %s\n\n", g.Conditional)
 	}
+	buf.Write(boilerplate)
```
[the remark this answers is not in the document]
Comment (krmayankk) [reply staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go:263]: should we create an issue to have some tests for these files ?

[the remark this answers is not in the document]
Comment (liggitt) [reply staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go:263]: with the fix to actually write to the checked in files, this bug was apparent (the copyright header wasn't present)

**Comment by dims** [comment]
/approve
/lgtm

that line did look suspicious :) thanks Jordan!

**Pull request merged**